rasa/telemetry.py

```diff
@@ -484,6 +509,56 @@ def toggle_telemetry_reporting(is_enabled: bool) -> None:
     rasa_utils.write_global_config_value(CONFIG_FILE_TELEMETRY_KEY, configuration)
 
 
+@ensure_telemetry_enabled
+def initialize_error_reporting() -> None:
+    """Sets up automated error reporting.
+
+    Exceptions are reported to sentry. We avoid sending any metadata (local
+    variables, paths, ...) to make sure we don't compromise any data. Only the
+    exception and its stacktrace is logged and only if the exception origins
+    from the `rasa` package. """
+    import sentry_sdk
+    from sentry_sdk.integrations.atexit import AtexitIntegration
+    from sentry_sdk.integrations.dedupe import DedupeIntegration
+    from sentry_sdk.integrations.excepthook import ExcepthookIntegration
+
+    # key for local testing can be found at
+    # https://sentry.io/settings/rasahq/projects/rasa-open-source/install/python/
+    # for local testing, set the key using `RASA_EXCEPTION_WRITE_KEY=key rasa <command>`
+    key = sentry_write_key()
+
+    if not key:
+        return
+
+    def strip_sensitive_data(event, hint):
+        # removes any paths from stack traces (avoids e.g. sending
+        # a users home directory name if package is installed there)
+        for value in event.get("exception", {}).get("values", []):
```

Comment:
let's also add a test for this. The functionality is not huge, but I think the test would be interesting as it would clearly show how an event typically looks like and what its structure is like.